### PR TITLE
Handle operations between surfaces of different topology

### LINKED ISSUE #665: Calculations between surfaces with different topology fails in `SurfaceViewerFMU`
It is often relevant to compare surfaces of different topology in FMU.
This should be supported

### PR DESCRIPTION
**Pull request by HansKallekleiv**
*Insert a description of your pull request (PR) here, and check off the boxes below when they are done.*

---

### Contributor checklist

- [ ] :tada: This PR closes #665.
- [ ] :scroll: I have broken down my PR into the following tasks:
   - [ ] Use xtgeo built in methods for operations between surfaces as they support different topologies
- [ ] :book: I have considered adding a new entry in `CHANGELOG.md`, and added it if should be communicated there.

### CODE REVIEW

**Comment by jcrivenaes** [comment]
This should already be handled autmatically in xtgeo if you take operations between instances:
Let s1 and s2 be two RegularSurface() instances, then e.g.

`diff = s1 - s2 `

shall regrid s2 to the topology of s1 behind the scene. See https://github.com/equinor/xtgeo/blob/ab3d188da03f100df65a62449473cccce5e43b5f/src/xtgeo/surface/_regsurf_oper.py#L44

**Comment by HansKallekleiv** [comment]
> This should already be handled autmatically in xtgeo if you take operations between instances:
> Let s1 and s2 be two RegularSurface() instances, then e.g.
> 
> `diff = s1 - s2 `
> 
> shall regrid s2 to the topology of s1 behind the scene. See https://github.com/equinor/xtgeo/blob/ab3d188da03f100df65a62449473cccce5e43b5f/src/xtgeo/surface/_regsurf_oper.py#L44

Thanks @jcrivenaes . Changed code to do operations between instances.

File: webviz_subsurface/plugins/_surface_viewer_fmu.py [hunk 1]
```diff
@@ -731,13 +731,12 @@ def calculate_surface_difference(
...
-        surface3.values = surface3.values + surface2.values
+        diff_surf = surface + surface2
     elif calculation == "Product":
-        surface3.values = surface3.values * surface2.values
+        diff_surf = surface * surface2
     elif calculation == "Quotient":
-        surface3.values = surface3.values / surface2.values
-    return surface3
+        diff_surf = surface / surface2
+    return diff_surf
```
Comment (anders-kiaer) [inline webviz_subsurface/plugins/_surface_viewer_fmu.py:742]: Call it something else than `diff_surf`? `calc_surf` or `result_surf`?

**Review by anders-kiaer — APPROVED** [review]
👍

**Pull request merged**